app/webpacker/components/RegistrationsV2/RegistrationAdministration/RegistrationAdministrationSearch.jsx

```diff
@@ -0,0 +1,104 @@
+import React, { useMemo } from 'react';
+import { Icon, Search } from 'semantic-ui-react';
+import _ from 'lodash';
+import useInputState from '../../../lib/hooks/useInputState';
+import useDebounce from '../../../lib/hooks/useDebounce';
+import { isoMoneyToHumanReadable } from '../../../lib/helpers/money';
+import { getStatusColor, getStatusIcon } from '../../../lib/utils/registrationAdmin';
+import { editRegistrationUrl } from '../../../lib/requests/routes.js.erb';
+import I18n from '../../../lib/i18n';
+
+const MIN_SEARCH_TEXT_LEN = 2;
+
+export default function RegistrationAdministrationSearch({
+  partitionedRegistrations,
+  competitionId,
+  usingPayments,
+  currencyCode,
+}) {
+  const [searchText, setSearchText] = useInputState('');
+  const debouncedSearchText = useDebounce(searchText);
+
+  const options = useMemo(
+    () => Object.fromEntries(
+      Object.entries(partitionedRegistrations).map(([status, registrations]) => [
+        status,
+        {
+          name: status,
+          results: registrations.map(({ user, competing, payment }) => ({
+            title: `${user.name}${user.wca_id ? ` (${user.wca_id})` : ''}`,
+            description: `${
+              user.email
+            }${
+              competing.comment
+                ? ` ${I18n.t('activerecord.attributes.registration.comments')}: ${competing.comment}`
+                : ''
+            }${
+              competing.admin_comment
+                ? ` ${I18n.t('activerecord.attributes.registration.administrative_notes')}: ${competing.admin_comment}`
+                : ''
+            }`,
+            price: usingPayments
+              ? isoMoneyToHumanReadable(payment.payment_amount_iso, currencyCode)
+              : undefined,
+            userId: user.id,
+            searchable: {
+              name: user.name,
+              wcaId: user.wca_id,
+              email: user.email,
+              comment: competing.comment,
+              note: competing.admin_comment,
+            },
+          })),
+        },
+      ]),
+    ),
```

Comment:
Lodash has `_.mapValues`, which may be an idiomatic alternative to what you're trying to do here

Reply:
I don't think that will work here because I need the key in the (new) value: `name: status`. This is used to get the icon when rendering the category. I don't see any other lodash functions which would work either.

The status could be found in any of the registrations, but there might be 0 registrations in the category (in which case we filter out the category, so I guess that wouldn't matter) and also we have different keys for the front-end and back-end (e.g. `waiting_list` vs `waiting` - see the `partitionRegistrations` function).

I'll probably do a follow-up PR here to clean up the naming and use the new icon/color helper functions (as you suggested in an earlier comment). But I can't directly align the keys them because we can't use snake case.